src/table/thead.tsx

```diff
@@ -89,12 +92,30 @@ const Thead = React.forwardRef(
 
     const { columnWidths, totalWidth, updateColumn } = useColumnWidths();
 
+    const stickyStyles = useStickyCellStyles({
+      stickyColumns: stickyState,
+      columnId: selectionColumnId.toString(),
+      getClassName: props => ({
+        [styles['sticky-cell']]: !!props,
+        [styles['sticky-cell-pad-left']]: !!props?.padLeft,
+        [styles['sticky-cell-last-left']]: !!props?.lastLeft,
+        [styles['sticky-cell-last-right']]: !!props?.lastRight,
```

Comment:
This is feedback for the actual hook and not directly this PR, but: I'm seeing this object/code block repeated quite a few times. Can this logic not be integrated into the hook somehow?

Reply:
We explicitly wanted to leave this out to the consumer because it has the `styles` dependency, which vary from cell type to cell type.